README
https://app.zeplin.io/project/5c7dcb5ab4e654bca8cde54d/screen/5cd0c46bce9a42346c709328


CHANGELOG.md
# v9.11.0 (Thu Nov 04 2021)

#### 🚀 Enhancement

- Pass through `className` on `option` element as child of select [#381](https://github.com/apollographql/space-kit/pull/381) ([@mayakoneval](https://github.com/mayakoneval))

#### Authors: 1

- [@mayakoneval](https://github.com/mayakoneval)

---

# v9.10.0 (Mon Oct 25 2021)

#### 🚀 Enhancement

- Source code icon [#380](https://github.com/apollographql/space-kit/pull/380) ([@jglovier](https://github.com/jglovier))

#### Authors: 1

- Joel Glovier ([@jglovier](https://github.com/jglovier))

---

# v9.9.0 (Wed Sep 22 2021)

#### 🚀 Enhancement

- Update tooltips to support controlled visibility [#379](https://github.com/apollographql/space-kit/pull/379) ([@Jephuff](https://github.com/Jephuff))

#### Authors: 1

- Jeffrey Burt ([@Jephuff](https://github.com/Jephuff))

---

# v9.8.0 (Wed Sep 15 2021)

#### 🚀 Enhancement

- Add a red asterisk to required FormLabels [#377](https://github.com/apollographql/space-kit/pull/377) ([@mayakoneval](https://github.com/mayakoneval))

#### Authors: 1

- [@mayakoneval](https://github.com/mayakoneval)

---

# v9.7.0 (Fri Sep 10 2021)

#### 🚀 Enhancement

- Modal: add xlarge size [#376](https://github.com/apollographql/space-kit/pull/376) ([@cy](https://github.com/cy))

#### Authors: 1

- Christine ([@cy](https://github.com/cy))

---

# v9.6.2 (Tue Aug 24 2021)

#### 🐛 Bug Fix

- modal - should be no title OR description on top margin [#374](https://github.com/apollographql/space-kit/pull/374) ([@cy](https://github.com/cy))

#### Authors: 1

- Christine ([@cy](https://github.com/cy))

---

# v9.6.1 (Tue Aug 24 2021)

#### 🐛 Bug Fix

- Modal: no title, no description dont render extra margin-top [#373](https://github.com/apollographql/space-kit/pull/373) ([@cy](https://github.com/cy))

#### Authors: 1

- Christine ([@cy](https://github.com/cy))

---

# v9.6.0 (Mon Aug 16 2021)

#### 🚀 Enhancement

- Fix border color in dark mode for AlertCard [#372](https://github.com/apollographql/space-kit/pull/372) ([@daniman](https://github.com/daniman))

#### Authors: 1

- Danielle Man ([@daniman](https://github.com/daniman))

---

# v9.5.0 (Fri Aug 13 2021)

#### 🚀 Enhancement

- Handle no table header specified case [#371](https://github.com/apollographql/space-kit/pull/371) ([@daniman](https://github.com/daniman))

#### Authors: 1

- Danielle Man ([@daniman](https://github.com/daniman))

---

# v9.4.0 (Fri Aug 13 2021)

#### 🚀 Enhancement

- Add dark theming to Card & Table components [#370](https://github.com/apollographql/space-kit/pull/370) ([@daniman](https://github.com/daniman))

#### Authors: 1

- Danielle Man ([@daniman](https://github.com/daniman))

---

# v9.3.1 (Tue Jun 15 2021)

#### 🐛  Bug Fix

- make dismissable optional: [#361](https://github.com/apollographql/space-kit/pull/361) ([@mayakoneval](https://github.com/mayakoneval))

#### Authors: 1

- [@mayakoneval](https://github.com/mayakoneval)

---

# v9.3.0 (Tue Jun 15 2021)

#### 🚀 Enhancement

- make alert card x optional [#360](https://github.com/apollographql/space-kit/pull/360) ([@mayakoneval](https://github.com/mayakoneval))

#### Authors: 1

- [@mayakoneval](https://github.com/mayakoneval)

---

# v9.2.0 (Fri Jun 11 2021)

#### 🚀 Enhancement

- Fix Popover component not being able to render >800 times without taking a dump on the main thread [#357](https://github.com/apollographql/space-kit/pull/357) ([@brainkim](https://github.com/brainkim))

#### Authors: 1

- Brian Kim ([@brainkim](https://github.com/brainkim))

---

# v9.1.0 (Tue Jun 08 2021)

#### 🚀 Enhancement

- Add icon for Sandbox [#356](https://github.com/apollographql/space-kit/pull/356) ([@jglovier](https://github.com/jglovier))

#### Authors: 1

- Joel Glovier ([@jglovier](https://github.com/jglovier))

---

# v9.0.0 (Wed Jun 02 2021)

#### 💥 Breaking Change

- Update themed components to respect theme from `SpaceKitProvider` [#352](https://github.com/apollographql/space-kit/pull/352) ([@daniman](https://github.com/daniman))

#### Authors: 1

- Danielle Man ([@daniman](https://github.com/daniman))

---

# v8.22.0 (Wed Jun 02 2021)

#### 🚀 Enhancement

- Add solid time icon [#353](https://github.com/apollographql/space-kit/pull/353) ([@esilverm](https://github.com/esilverm))

#### Authors: 1

- Evan Silverman ([@esilverm](https://github.com/esilverm))

---

# v8.21.1 (Wed Jun 02 2021)

#### 🐛 Bug Fix

- Enable versionBranches, document backport releases [#351](https://github.com/apollographql/space-kit/pull/351) ([@glasser](https://github.com/glasser))

#### Authors: 1

- David Glasser ([@glasser](https://github.com/glasser))

---

# v8.21.0 (Fri May 21 2021)

#### 🚀 Enhancement

- Only remove ConfirmationTooltip timeout when unmounting [#348](https://github.com/apollographql/space-kit/pull/348) ([@jgzuke](https://github.com/jgzuke))

#### Authors: 1

- Jason Zukewich ([@jgzuke](https://github.com/jgzuke))

---

# v8.20.2 (Fri Apr 23 2021)

#### 🐛  Bug Fix

- Use onMouseDown instead of onClick for auto closing modals [#342](https://github.com/apollographql/space-kit/pull/342) ([@Jephuff](https://github.com/Jephuff))

#### Authors: 1

- Jeffrey Burt ([@Jephuff](https://github.com/Jephuff))

---

# v8.20.1 (Mon Apr 19 2021)

#### 🐛  Bug Fix

- update peer dependencies versions [#340](https://github.com/apollographql/space-kit/pull/340) ([@Jephuff](https://github.com/Jephuff))

#### Authors: 1

- Jeffrey Burt ([@Jephuff](https://github.com/Jephuff))

---

# v8.20.0 (Thu Apr 15 2021)

#### 🚀 Enhancement

- Add streamline megaphone icon [#339](https://github.com/apollographql/space-kit/pull/339) ([@jgzuke](https://github.com/jgzuke))

#### Authors: 1

- Jason Zukewich ([@jgzuke](https://github.com/jgzuke))

---

# v8.19.1 (Tue Mar 23 2021)

#### 🐛  Bug Fix

- Pass maxWidth to Popover in Select [#334](https://github.com/apollographql/space-kit/pull/334) ([@Jephuff](https://github.com/Jephuff))

#### Authors: 1

- Jeffrey Burt ([@Jephuff](https://github.com/Jephuff))

---


Button.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var tslib_1 = require("tslib");
var core_1 = require("@emotion/core");
var colors = tslib_1.__importStar(require("./colors"));
var typography_1 = require("./typography");
exports.Button = function (_a) {
    var _b = _a.backgroundColor, backgroundColor = _b === void 0 ? colors.silver.light : _b, children = _a.children, _c = _a.disabled, disabled = _c === void 0 ? false : _c, _d = _a.disabledBackgroundColor, disabledBackgroundColor = _d === void 0 ? colors.silver.dark : _d, _e = _a.disabledTextColor, disabledTextColor = _e === void 0 ? colors.grey.light : _e, _f = _a.fab, fab = _f === void 0 ? false : _f, _g = _a.forceActiveState, forceActiveState = _g === void 0 ? false : _g, _h = _a.forceFocusState, forceFocusState = _h === void 0 ? false : _h, _j = _a.forceHoverState, forceHoverState = _j === void 0 ? false : _j, _k = _a.hidden, hidden = _k === void 0 ? false : _k, _l = _a.hiddenActiveBackgroundColor, hiddenActiveBackgroundColor = _l === void 0 ? colors.silver.dark : _l, _m = _a.hiddenFocusedColor, hiddenFocusedColor = _m === void 0 ? colors.blue.base : _m, _o = _a.hoverBackgroundColor, hoverBackgroundColor = _o === void 0 ? colors.silver.base : _o, icon = _a.icon, _p = _a.size, size = _p === void 0 ? "default" : _p, otherProps = tslib_1.__rest(_a, ["backgroundColor", "children", "disabled", "disabledBackgroundColor", "disabledTextColor", "fab", "forceActiveState", "forceFocusState", "forceHoverState", "hidden", "hiddenActiveBackgroundColor", "hiddenFocusedColor", "hoverBackgroundColor", "icon", "size"]);
    var iconSize = size === "small" ? 12 : size === "large" ? 24 : 16;
    var iconOnly = !children;
    if (fab) {
        if (!icon) {
            throw new TypeError("FAB buttons are required to have an `icon`");
        }
        else if (children) {
            throw new TypeError("FAB buttons cannot have children, only an `icon`");
        }
    }
    if (forceActiveState && forceHoverState) {
        throw new TypeError("Do not force multiple properties at once, you will get indeterminiate behavior");
    }
    var activeProperties = tslib_1.__assign({}, (hidden
        ? { backgroundColor: hiddenActiveBackgroundColor }
        : {
            boxShadow: "inset 0 0 0 1px rgba(18, 21, 26, 0.2), inset 0 -1px 0 0 rgba(18, 21, 26, 0.05), inset 0 2px 2px 0 rgba(18, 21, 26, 0.12)"
        }), { outline: 0 });
    var hoverProperties = tslib_1.__assign({ backgroundColor: hoverBackgroundColor, cursor: "pointer" }, (!hidden && {
        boxShadow: "0 5px 10px 0 rgba(18, 21, 26, 0.12), inset 0 0 0 1px rgba(18, 21, 26, 0.2), inset 0 -1px 0 0 rgba(18, 21, 26, 0.05)"
    }));
    var focusProperties = tslib_1.__assign({ boxShadow: "0 1px 4px 0 rgba(18, 21, 26, 0.08), 0 0 0 2px #bbdbff, inset 0 0 0 1px #2075d6, inset 0 -1px 0 0 rgba(18, 21, 26, 0.05)" }, (hidden && { color: hiddenFocusedColor + " !important" }));
    return (core_1.jsx("button", tslib_1.__assign({}, otherProps, { css: tslib_1.__assign({ appearance: "button", backgroundColor: hidden
                ? "transparent"
                : disabled
                    ? disabledBackgroundColor
                    : backgroundColor, borderRadius: fab ? "100%" : 4, borderWidth: 0 }, (!hidden && {
            boxShadow: "0 1px 4px 0 rgba(18, 21, 26, 0.08), inset 0 0 0 1px rgba(18, 21, 26, 0.2), inset 0 -1px 0 0 rgba(18, 21, 26, 0.05)"
        }), { minWidth: iconOnly
                ? size === "small"
                    ? 28
                    : size === "large"
                        ? 42
                        : 36
                : size === "small"
                    ? 76
                    : size === "large"
                        ? 112
                        : 100, padding: size === "small" ? "5px 8px" : size === "large" ? 8 : 7 }, (size === "small"
            ? typography_1.base.small
            : size === "large"
                ? typography_1.base.large
                : typography_1.base.base), (!disabled &&
            (forceHoverState
                ? hoverProperties
                : {
                    ":hover": hoverProperties
                })), { fontWeight: 600, outline: 0 }, (disabled && { color: disabledTextColor + " !important" }), (!disabled &&
            (forceFocusState ? focusProperties : { ":focus": focusProperties })), (!disabled && forceActiveState
            ? activeProperties
            : { ":active": activeProperties })) }),
        core_1.jsx("div", { css: {
                display: "flex",
                alignItems: "center",
                justifyContent: "center"
            } },
            icon && (core_1.jsx("span", { css: {
                    display: "inline-block",
                    width: iconSize,
                    height: iconSize,
                    margin: iconOnly ? "3px 0" : "0 4px 0"
                } }, icon)),
            children && core_1.jsx("span", null, children))));
};
//# sourceMappingURL=Button.js.map